app/services/character/relocator.rb

```diff
@@ -0,0 +1,115 @@
+# Steps:
+# validate single user, unshared icons, galleries, and templates (if included), lack of character groups
+# collect: templates (if included), characters, posts, replies, drafts, galleries, icons
+# update all on unaudited (templates, aliases, drafts, galleries, icons)
+# UpdateModelJob on audited (characters, posts, replies)
+
+class Character::Relocator < Object
+  def initialize(admin_id)
+    @audited_user_id = admin_id
+  end
+
+  def transfer(character_ids, new_user_id, include_templates: false)
+    @characters = Character.where(id: character_ids)
+    @user_id = validate_user!
+    @user_other_characters = Character.where(user_id: @user_id).where.not(id: @characters.ids)
+    @other_character_galleries = get_galleries(@user_other_characters)
+    @other_character_icon_ids = get_icon_ids(@user_other_characters, @other_character_galleries)
+
+    raise CharacterGroupError.new('Characters must not have groups') if @characters.where.not(character_group: nil).exists?
+    @galleries = validate_galleries!
+    @icons = validate_icons!
+    @templates = validate_templates! if include_templates
+
+    @posts = Post.where(character_id: character_ids, user_id: @user_id)
+    @replies = Reply.where(character_id: character_ids, user_id: @user_id)
+    @drafts = ReplyDraft.where(character_id: character_ids, user_id: @user_id)
+
+    User.find(new_user_id)
+
+    @new_authors = calc_new_authors(new_user_id)
+    @rem_authors = calc_rem_authors
+
+    # rubocop:disable Rails/SkipsModelValidations
+    @templates.update_all(user_id: new_user_id) if include_templates
+    @galleries.update_all(user_id: new_user_id)
+    @icons.update_all(user_id: new_user_id)
+    @drafts.update_all(user_id: new_user_id)
+    # rubocop:enable Rails/SkipsModelValidations
+
+    UpdateModelJob.perform_later(Post.to_s, { character_id: @characters.ids }, { user_id: new_user_id }, @audited_user_id)
+    UpdateModelJob.perform_later(Reply.to_s, { character_id: @characters.ids }, { user_id: new_user_id }, @audited_user_id)
+    UpdateModelJob.perform_later(Character.to_s, { id: @characters.ids }, { user_id: new_user_id }, @audited_user_id)
+
+    @new_authors.each { |post_id| Post::Author.create!(post_id: post_id, user_id: new_user_id) }
+    @rem_authors.each { |post_id| Post::Author.find_by!(post_id: post_id, user_id: @user_id).destroy! }
+  end
+
+  private
+
+  def validate_user!
+    user_id = @characters.select(:user_id).distinct.pluck(:user_id)
+    raise RequireSingleUser.new('Characters must all have the same original user') if user_id.length > 1
+    user_id = user_id.first
```

Comment:
```suggestion
    User.find(user_id.first).id
```

Reply:
no, that doesn't work b/c then you -- well, you'd have to put the first on the return, too, I guess?

Reply:
This should be able to replace this and the following couple lines

Reply:
-- oh, because you're asking for the id off it -- yeah, that could work